GVFS/GVFS/CommandLine/SparseVerb.cs

```diff
@@ -166,6 +166,13 @@ protected override void Execute(GVFSEnlistment enlistment)
                         {
                             needToChangeProjection = true;
                             foldersToRemove.AddRange(directories);
+                            directories.Clear();
+                        }
+
+                        // Need to add folders that will be in the projection back into directories for the status check
+                        foreach (string folder in foldersToAdd)
+                        {
+                            directories.Add(folder);

```

Comment:
Do we also need to add pack in directories that were in the set before the command was run?

Consider this scenario:

- InitialSet: { A, B, C }
- `set` is called with { B, C, D }

`foldersToRemove ` will be { A } and `foldersToAdd` will be { D } meaning that after this loop `directories` will only have { D }, when it should be { B, C, D }.

Rather than re-adding `foldersToAdd` should we be re-adding `folders`?

If there's not already it might be worth adding a test that would catch this.

Reply:
To clarify, I think the problem would be if something was modified in { B, C } git status would flag it when it probably shouldn't.

Reply:
I will add a test case for this.